File: ws/vizcaino/u3/ws29Singleton/wsSingleton/src/main/java/ec/edu/espe/wssingleton/Calculator.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 */

package ec.edu.espe.wssingleton;

/**
 *
 * @author Adrian Vizcaino <The-Softwarriors at ESPE>
 */
public class Calculator {

    public static void main(String[] args) {
        UsTax taxService = UsTax.getInstance();

        double saleAmount = 100.0;
        double tax = taxService.calculateTax(saleAmount);
        double total = saleAmount + tax;

        System.out.println("Sale Amount: $" + saleAmount);
        System.out.println("Applied Tax Rate: " + (taxService.getTaxRate() * 100) + "%");
        System.out.println("Calculated Tax: $" + tax);
        System.out.println("Total Sale: $" + total);
    }
}

File: hw/vizcaino/u3/hw23Singleton/hw23SingletonJavaScript/calculator.js
const USTax = require('./usTax');

class Calculator {
    static main() {
        const taxService = USTax.getInstance();

        const saleAmount = 100.0;
        const tax = taxService.calculateTax(saleAmount);
        const total = saleAmount + tax;

        console.log(`Sale Amount: $${saleAmount}`);
        console.log(`Applied Tax Rate: ${taxService.getTaxRate() * 100}%`);
        console.log(`Calculated Tax: $${tax}`);
        console.log(`Total Sale: $${total}`);
    }
}

Calculator.main();